src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,573 @@
+        """
+        Compute optimal scale and resolutions for a point cloud or a distance matrix.
+
+        Parameters:
+            X (numpy array of shape (num_points) x (num_coordinates) if point cloud and (num_points) x (num_points) if distance matrix): input point cloud or distance matrix.
+            beta (double): exponent parameter (default 0.). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            C (double): constant parameter (default 10.). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            N (int): subsampling iterations (default 100). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+
+        Returns:
+            delta (double): optimal scale that can be used with agglomerative clustering.
+            resolutions (numpy array of shape (num_filters): optimal resolutions associated to each filter.
+        """
+        num_filt, delta = self.filters.shape[1], 0
+        delta = self.estimate_scale(X=X, N=N, inp=self.input_type, C=C, beta=beta)
+
+        pairwise = pairwise_distances(X, metric="euclidean") if self.input_type == "point cloud" else X
+        pairs = np.argwhere(pairwise <= delta)
+        num_pairs = pairs.shape[0]
+        res = []
+        for f in range(num_filt):
+            F = self.filters[:,f]
+            resf = 0
+            for p in range(num_pairs):
+                resf = max(resf, abs(F[pairs[p,0]] - F[pairs[p,1]]))
+            res.append(resf)
+
+        return delta, np.array(res)
+
+    def fit(self, X, y=None):
+        """
+        Fit the MapperComplex class on a point cloud or a distance matrix: compute the Mapper complex and store it in a simplex tree called simplex_tree.
+
+        Parameters:
+            X (numpy array of shape (num_points) x (num_coordinates) if point cloud and (num_points) x (num_points) if distance matrix): input point cloud or distance matrix.
+            y (n x 1 array): point labels (unused).
+        """
+        if self.filters is None:
+            if self.input_type == "point cloud":
+                self.filters = X[:,0:1]
+            elif self.input_type == "distance matrix":
+                self.filters = X.max(axis=0)[:,None]
+        if self.colors is None:
+            if self.input_type == "point cloud":
+                self.colors = X[:,0:1]
+            elif self.input_type == "distance matrix":
+                self.colors = X.max(axis=0)[:,None]
+
+        num_pts, num_filters = self.filters.shape[0], self.filters.shape[1]
+
+        # If some filter limits are unspecified, automatically compute them
+        if self.filter_bnds is None:
+            self.filter_bnds = np.hstack([np.min(self.filters, axis=0)[:,np.newaxis], np.max(self.filters, axis=0)[:,np.newaxis]])
+
+        # If some resolutions are not specified, automatically compute them
+        if self.gains is None:
+            self.gains = .33 * np.ones(num_filters)
```

Comment:
Should we say something about the default value in the doc? Or say that some unspecified default is used?

Reply:
I've added a small comment about this in the doc.